restapi/admin_users.go

```
@@ -467,3 +474,50 @@ func getAddUsersListToGroupsResponse(session *models.Principal, params admin_api
 
 	return nil
 }
+
+func getListUsersWithAccessToBucketResponse(session *models.Principal, bucket string) ([]string, *models.Error) {
+	ctx := context.Background()
+	mAdmin, err := newMAdminClient(session)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	// create a minioClient interface implementation
+	// defining the client to be used
+	adminClient := adminClient{client: mAdmin}
+
+	users, err := listUsers(ctx, adminClient)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	var retval []string
+	seen := make(map[string]bool)
+	for i := 0; i < len(users); i++ {
+		policy, err := adminClient.getPolicy(ctx, users[i].Policy)
+		if err == nil {
+			parsedPolicy, err2 := parsePolicy(users[i].Policy, policy)
+			if err2 == nil && policyMatchesBucket(parsedPolicy, bucket) {
+				retval = append(retval, users[i].AccessKey)
+				seen[users[i].AccessKey] = true
+			}
+		}
+	}
+
+	groups, _ := listGroups(ctx, adminClient)
```

we should not ignore this error, in case groups is `nil` the `len(* groups)`  will panic with: `panic: runtime error: invalid memory address or nil pointer dereference`